src/account/AccountFactory.sol

```diff
@@ -0,0 +1,86 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.19;
+
+import {IEntryPoint} from "@eth-infinitism/account-abstraction/interfaces/IEntryPoint.sol";
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ERC1967Proxy} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Proxy.sol";
+import {Create2} from "@openzeppelin/contracts/utils/Create2.sol";
+
+import {UpgradeableModularAccount} from "../../src/account/UpgradeableModularAccount.sol";
+import {ValidationConfigLib} from "../../src/helpers/ValidationConfigLib.sol";
+import {SingleSignerValidation} from "../../src/plugins/validation/SingleSignerValidation.sol";
+
+contract AccountFactory is Ownable {
+    UpgradeableModularAccount public accountImplementation;
```

Comment:
I think we should make this immutable and just deploy new factories for new versions, this way deployment saves an SLOAD, wdyt?

Reply:
bump, if we don't support upgrading this, we should make it immutable

Reply:
whoops, missed it, just made the change